scipy/optimize/tests/test_optimize.py

```diff
@@ -177,6 +177,7 @@ def test_bfgs(self):
                          [0, -5.24885582e-01, 4.87530347e-01]],
                         atol=1e-14, rtol=1e-7)
 
+    @pytest.mark.filterwarnings('ignore::UserWarning')
```

Comment:
If the test always generates a warning, this should be replaced with `with pytest.warns(OptimizeWarning):`.

Reply:
It doesn't.